File: server/CampaignTracker.Models/Entities/SessionEvent.cs
namespace CampaignTracker.Models;

public class SessionEvent : EntityBase
{
    public int Index { get; set; }
    public int SessionId { get; set;}
    public int EventTypeId { get; set; }
    public int TotalDuration { get; set; }
    public string Title { get; set; }
    public string Notes { get; set; }

    public Session Session { get; set; }
    public EventType EventType { get; set; }
    public IEnumerable<Character> Characters { get; set; }
    public IEnumerable<Duration> Durations { get; set; }
}

File: client/core/models/api/session-event.ts
import { EntityBase } from "../entity-base";
import { Session } from "./session";
import { Character } from "./character";
import { Duration } from "./duration";
import { EventType } from "./eventType";
import { FormBuilder, FormGroup, Validators } from "@angular/forms";

export interface SessionEvent extends EntityBase {
    index?: number;
    sessionId: number;
    eventTypeId?: number;
    totalDuration?: number;
    title: string;
    notes: string;

    session: Session;
    eventType: EventType;
    characters: Character[];
    durations: Duration[];   
}

export const GenerateEventForm =  (event: SessionEvent, fb: FormBuilder): FormGroup => 
    fb.group({
        id: [event?.id],
        index: [event?.index],
        sessionId: [event?.sessionId, Validators.required],
        eventTypeId: [event?.eventTypeId, Validators.required],
        totalDuration: [event?.totalDuration],
        title: [event?.title, Validators.required],
        notes: [event?.notes]
    })